### PR TITLE
CAR-17: Add support for asserting JSONs equality in tests

### PR DESCRIPTION
**Pull request by majagabajaga**
- Added support for loading JSONs as JsonElement from resources
- Added Asserting of equality of JSONs
- Added UTF-8 as default encoding
- Added commons-io as dependencies

### CODE REVIEW

File: .gitattributes [hunk 1]
```diff
@@ -0,0 +1 @@
+*.txt	eof=lf
```
Comment (mironuk) [inline .gitattributes]: There shoud be new line after last line.

File: src/test/java/carservice/test/AssertExt.java [hunk 2]
```diff
@@ -13,37 +13,36 @@
...
-    public static void assertJsonFilesEqual(Class<?> clazz, String path1, String path2) {
-        Reader r1 = TestUtils.getReader(clazz, path1);
-        Reader r2 = TestUtils.getReader(clazz, path2);
+    public static void assertJsonFilesEqual(Class<?> clazz, String expectedJsonPath, String actualJsonPath) {
+        Reader r1 = TestUtils.getReader(clazz, expectedJsonPath);
+        Reader r2 = TestUtils.getReader(clazz, actualJsonPath);
 
-        JsonReader jsonReader1 = new JsonReader(r1);
-        JsonReader jsonReader2 = new JsonReader(r2);
+        JsonReader expectedJsonReader = new JsonReader(r1);
```
Comment (mironuk) [inline src/test/java/carservice/test/AssertExt.java]: Seems like these two lines can be moved to common method also.

File: src/test/java/carservice/test/testdata/f1.json [hunk 3]
```diff
@@ -1,3 +1,8 @@
 {
-    "meow": "meow"
+    "meow": "meow",
+    "data": {
+        "color": "black",
+        "meowType": "annoying"
+    },
+    "cat": "Barseeq"
```
Comment (mironuk) [inline src/test/java/carservice/test/testdata/f1.json:7]: Show stopper error: "Barseeq" should be "Barseek". Just kidding. :)

File: src/test/java/carservice/test/AssertExtTest.java [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+    @Test(expected = ComparisonFailure.class)
+    public void givenUnequalJsons_whenAssertJsonFilesEqual_thenComparisonFailure() {
+        AssertExt.assertJsonFilesEqual(AssertExtTest.class, "f1.json", "f3.json");
+    }
+
+    @Test
+    public void givenEqualJsons_whenAssertJsonsEqual_thenSuccess() {
+        // Arrange
+        String json1 = "{ \"meow\": \"meow\" }";
```
Comment (mironuk) [inline src/test/java/carservice/test/AssertExtTest.java]: I think there should be string examples from f1.json and f2.json (full json with shuffled fields)

**Review by mironuk — CHANGES_REQUESTED** [review]

File: .gitattributes (hunk 1 above)
Comment (majagabajaga) [reply .gitattributes]: Added the new line character.

File: src/test/java/carservice/test/AssertExt.java (hunk 2 above)
Comment (majagabajaga) [reply src/test/java/carservice/test/AssertExt.java]: Done.

File: src/test/java/carservice/test/testdata/f1.json (hunk 3 above)
Comment (majagabajaga) [reply src/test/java/carservice/test/testdata/f1.json:7]: Donen't.

File: src/test/java/carservice/test/AssertExtTest.java (hunk 4 above)
Comment (majagabajaga) [reply src/test/java/carservice/test/AssertExtTest.java]: Done.

**Pull request merged**